Updated Domains success screen

frosty opened the pull request:
This PR implements the success screen for the domains purchase flow. When rotating to landscape on a phone I opted to remove the illustration. When text gets very big, the illustration may be hidden (for 'accessibility' type sizes) and the view should scroll.

cc @kyleaparker

| Standard | XXL | Accessibility XXXL |
|---|---|---|
| ![Simulator Screen Shot - iPhone 12 mini - 2021-10-14 at 16 15 00](https://user-images.githubusercontent.com/4780/137350824-d0b68cfc-bc4a-40fa-ba5c-9bbf968b0a82.png) | ![Simulator Screen Shot - iPhone 12 mini - 2021-10-14 at 15 02 20](https://user-images.githubusercontent.com/4780/137351078-658d03ae-806a-46d8-9516-bbd4a679fa3a.png) | ![Simulator Screen Shot - iPhone 12 mini - 2021-10-14 at 15 02 33](https://user-images.githubusercontent.com/4780/137350993-70b411bc-b027-4101-a577-196d7041f272.png) |
| ![Simulator Screen Shot - iPhone 12 mini - 2021-10-14 at 16 15 03](https://user-images.githubusercontent.com/4780/137350802-8c87a815-a8f3-4946-99ef-7871ba7cc3f7.png) | ![Simulator Screen Shot - iPhone 12 mini - 2021-10-14 at 15 02 23](https://user-images.githubusercontent.com/4780/137351067-7d07a6cb-0add-4965-bd84-6c8aed776be5.png) | ![Simulator Screen Shot - iPhone 12 mini - 2021-10-14 at 15 02 38](https://user-images.githubusercontent.com/4780/137350962-2d9e327b-fbc1-4c7b-9fad-ca56dfa77c4c.png) |

## To Test

- Build and run
- Try both of the following:
    - Redeem a domain credit to use the native flow, and check you see this screen at the end
    - Add another domain to use the web flow and check you see this screen. I was unable to test this as I'm having issues again with the webview auto dismissing. Looking at the code, I _think_ it should be displayed though.
- Check with different dynamic font sizes

## Regression Notes

1. Potential unintended areas of impact

- This affects the existing domain flows, but this screen is an improvement over what we had already. Changes are limited to this new screen.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

frosty committed:
Update style of domain purchase success screen to match designs

frosty committed:
Hide imageview when compact height or using accessibility content size

frosty committed:
Allow domain success screen content to scroll if text gets too large

frosty committed:
Hound fix

frosty commented on WordPress/Classes/ViewRelated/Domains/Domain credit/DomainCreditRedemptionSuccessViewController.swift:
```diff
@@ -1,12 +1,15 @@
 import UIKit
+import WordPressAuthenticator
```
I'm not a fan of using Authenticator here for the NUXButtonViewController, but we've already used it elsewhere in the domains flow so I've done it here for consistency of UI and ease of implementation. We should pull this out into the main app soon.

Gio2018 replied:
Can we use a `FancyButton` instead, similarly to what we are doing in `WhatIsNewView.swift` for the `continueButton`?

mokagio commented:
I'm going to bump this to the next release because we'll be code freezing 18.5 today and it seems like this needs a bit more touches.

From the looks of it, it seems like this might be something we want in 18.5, though? If this cannot wait two weeks and it's important that it makes it into this release, let me know and we'll organize a new beta once ready. 👍 😃

Gio2018 commented on WordPress/Classes/ViewRelated/Domains/Domain credit/DomainCreditRedemptionSuccessViewController.swift:
```diff
@@ -20,50 +23,168 @@ class DomainCreditRedemptionSuccessViewController: UIViewController {
+            buttonContainer.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            buttonContainer.heightAnchor.constraint(greaterThanOrEqualToConstant: Metrics.buttonControllerMinHeight)
+        ])
     }
```
Any chance we can break this `viewDidLoad` into smaller methods?

diegoreymendez replied:
+1

frosty replied:
👍

Gio2018 reviewed (changes requested):
Hey @frosty , the test cases worked as expected (including the web view), but I noticed that if the feature flag is off, it still shows the new success screen, and the same happens with the domain prices (cc @diegoreymendez ).
Other than that, I left a couple of code comments.

diegoreymendez commented on WordPress/Classes/ViewRelated/Domains/Domain credit/DomainCreditRedemptionSuccessViewController.swift:
```diff
@@ -20,50 +23,168 @@ class DomainCreditRedemptionSuccessViewController: UIViewController {
+
+        let illustration = UIImageView(image: UIImage(named: "domains-success"))
+        illustration.contentMode = .scaleAspectFit
+        self.illustration = illustration
```
We could initialize the `illustration` property lazily where it's declared instead of here since the initialization doesn't really take any external parameter.

If we did that the `illustration` property would no longer be optional and this method (`viewDidLoad`) would be simplified a bit.

frosty replied:
👍

kyleaparker commented:
Looking good @frosty !

I think we can remove the shadow from behind the button area, we generally use a separator line elsewhere in the app. 

I'm not sure if you have worked on the dark mode version yet, if not this is what I had in mind: 
![1  Overlay_ Purchase complete](https://user-images.githubusercontent.com/712873/137916485-5fd0e585-2944-4ee2-ba4e-37258306859c.png)

It's using the secondary system background colour and light blue for the primary button. 

Let me know what you think!

diegoreymendez reviewed (commented):
Adding some comments on the code.

frosty committed:
Switch NUXButtonViewController for a FancyButton

frosty committed:
Refactor viewDidLoad

frosty committed:
Domains success screen: Improved dark mode appearance

frosty commented:
Thanks folks, I think I've addressed all the feedback now. I've improved the appearance in dark mode too.

@Gio2018 The only thing I haven't done is use a feature flag – as this is just a visual improvement over the previous screen, I didn't feel that it really needed feature flagging, but if you think otherwise I can duplicate the previous view controller and selectively display the 'correct' screen according to the feature flag.

frosty commented:
@kyleaparker How's this?

|   |   |
|---|---|
| ![Simulator Screen Shot - iPhone 12 Pro - 2021-10-20 at 09 43 10](https://user-images.githubusercontent.com/4780/138064081-4ac64181-1280-4605-a8a4-08c149cbf951.png) | ![Simulator Screen Shot - iPhone 12 Pro - 2021-10-20 at 10 04 43](https://user-images.githubusercontent.com/4780/138064083-2dd192b7-536a-4c39-a489-98380179e4f2.png) |

kyleaparker commented:
Looks great!
